AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -3657,6 +3658,37 @@ function Debug-AzStorageAccountAuth {
             }
         }
 
+        if (!$filterIsPresent -or $Filter -match "CheckDefaultSharePermission")
+        {
+            try {
+                $checksExecuted += 1
+                Write-Verbose "CheckDefaultSharePermission - START"
+
+                $StorageAccountObject = Validate-StorageAccount `
+                    -ResourceGroupName $ResourceGroupName `
+                    -StorageAccountName $StorageAccountName `
+                    -ErrorAction Stop
+ 
+                Write-Verbose -Message "Storage account $StorageAccountName is already joined in domain $($activeDirectoryProperties.DomainName)."
+                
+                $DefaultSharePermission = $StorageAccountObject.AzureFilesIdentityBasedAuth.DefaultSharePermission
+                
+                # If DefaultSharePermission is null or 'None'
+                if((!$DefaultSharePermission) -or ($DefaultSharePermission -eq 'None')){
+                    Write-Error "CheckDefaultSharePermission - FAILED"

```

Comment:
DefaultSharePermission is optional, so missing it is not an error. Use Write-Verbose, and add a link to https://docs.microsoft.com/en-us/azure/storage/files/storage-files-identity-ad-ds-assign-permissions?tabs=azure-portal, so that customer can follow up to configure it if needed.